bob-backend/src/pearl/group.rs

```diff
@@ -568,33 +589,54 @@ impl Group {
         });
     }
 
-    pub async fn delete(&self, key: BobKey) -> Result<u64, Error> {
-        let holders = self.holders.read().await;
+    pub async fn delete(
+        &self,
+        key: BobKey,
+        is_alien: bool,
+        timestamp_config: StartTimestampConfig,
+    ) -> Result<u64, Error> {
+        let create_new = {
+            let holders = self.holders.read().await;
+            holders.len() == 0
+        };
+        if create_new {
+            let holder = self
+                .get_actual_holder(get_current_timestamp(), timestamp_config)
+                .await?
+                .1;
+            Self::delete_in_holders(std::iter::once(&holder), key, is_alien).await
+        } else {
+            let holders = self.holders.read().await;
+            Self::delete_in_holders(holders.iter(), key, is_alien).await
+        }
+    }
+
+    async fn delete_in_holders(
+        holders: impl Iterator<Item = &Holder>,
+        key: BobKey,
+        is_alien: bool,
+    ) -> Result<u64, Error> {
         let mut total_count = 0;
-        for holder in holders.iter() {
-            let delete = Self::delete_common(holder.clone(), key).await;
+        for holder in holders {
+            let delete = Self::delete_common(holder.clone(), key, is_alien).await;
             match delete {
                 Ok(count) => {
                     trace!("delete data: {:?} from: {:?}", count, holder);
                     total_count += count;
                 }
                 Err(err) => {
-                    if err.is_key_not_found() {
-                        debug!("{} not found in {:?}", key, holder)
-                    } else {
-                        error!("delete error: {}, from : {:?}", err, holder);
-                        return Err(err);
-                    }
+                    error!("delete error: {}, from : {:?}", err, holder);
+                    return Err(err);
                 }
             }
         }
         Ok(total_count)
     }
 
-    async fn delete_common(holder: Holder, key: BobKey) -> Result<u64, Error> {
-        let result = holder.delete(key).await;
+    async fn delete_common(holder: Holder, key: BobKey, is_alien: bool) -> Result<u64, Error> {
+        let result = holder.delete(key, is_alien).await;
```

Comment:
Please, handle `is_possible_disk_disconnection` in the same way as in `put_common`